packages/web/titanium/hacks/dialog-navigation-hack.ts

```diff
@@ -0,0 +1,13 @@
+import { MdDialog } from '@material/web/dialog/dialog';
+
+/** 
+    Adds a listener to the window popstate event that will close the dialog when the user navigates away from the page.
+**/
+
+export function dialogNavigationHack(dialog: MdDialog) {
+  const closeListener = () => {
+    dialog.close('navigation-close');
+    window.removeEventListener('popstate', closeListener, false);
```

Comment:
Would have to remove the listener on close too.  Also should check is dialog is open prior to calling close?